editor/window_wrapper.cpp

```diff
@@ -0,0 +1,397 @@
+
+	enable_window_on_screen(p_screen, false);
+	window->set_position(p_rect.position);
+	window->set_size(p_rect.size);
+}
+
+void WindowWrapper::enable_window_on_screen(int p_screen, bool p_auto_scale) {
+	int current_screen = Object::cast_to<Window>(get_viewport())->get_current_screen();
+
+	bool auto_scale = p_auto_scale && !EDITOR_GET("interface/multi_window/maximize_window");
+
+	if (auto_scale && current_screen != p_screen) {
+		Rect2 control_rect = _get_default_window_rect();
+
+		Rect2i source_screen_rect = DisplayServer::get_singleton()->screen_get_usable_rect(current_screen);
+		Rect2i dest_screen_rect = DisplayServer::get_singleton()->screen_get_usable_rect(p_screen);
+
+		// Adjust the window rect size in case the resolution changes.
+		Vector2 screen_ratio = Vector2(source_screen_rect.size) / Vector2(dest_screen_rect.size);
+
+		// The screen positioning may change, so remove the original screen position.
+		control_rect.position -= source_screen_rect.position;
+		control_rect = Rect2i(control_rect.position * screen_ratio, control_rect.size * screen_ratio);
+		control_rect.position += dest_screen_rect.position;
+
+		restore_window(control_rect, p_screen);
+	} else {
+		set_window_enabled(true);
+		window->set_current_screen(p_screen);
+	}
+}
+
+void WindowWrapper::set_window_title(const String p_title) {
+	if (!is_window_available()) {
+		return;
+	}
+	window->set_title(p_title);
+}
+
+void WindowWrapper::set_margins_enabled(bool p_enabled) {
+	ERR_FAIL_COND(get_window_enabled());
+	if (bool(margins != nullptr) == p_enabled) {
+		return;
+	}
+
+	if (margins) {
+		margins->queue_free();
+		margins = nullptr;
+	} else if (p_enabled) {
+		Size2 borders = Size2(4, 4) * EDSCALE;
+		margins = memnew(MarginContainer);
+		margins->add_theme_constant_override("margin_right", borders.width);
+		margins->add_theme_constant_override("margin_top", borders.height);
+		margins->add_theme_constant_override("margin_left", borders.width);
+		margins->add_theme_constant_override("margin_bottom", borders.height);
+
+		window->add_child(margins);
+		margins->set_anchors_and_offsets_preset(PRESET_FULL_RECT);
+	}
+}
+
+WindowWrapper::WindowWrapper() {
+	if (SceneTree::get_singleton()->get_root()->is_embedding_subwindows() && EDITOR_GET("interface/multi_window/enable")) {
+		return;
+	}
+
+	window = memnew(Window);
+	window->set_wrap_controls(true);
+
+	add_child(window);
+	window->hide();
+
+	window->connect("close_requested", callable_mp(this, &WindowWrapper::set_window_enabled).bind(false));
+
+	ShortcutBin *capturer = memnew(ShortcutBin);
+	window->add_child(capturer);
+
+	window_background = memnew(Panel);
+	window_background->set_anchors_and_offsets_preset(PRESET_FULL_RECT);
+	window->add_child(window_background);
+
+	set_process_shortcut_input(true);
+
+	EDITOR_DEF_RST("interface/multi_window/enable", true);
+	EDITOR_DEF("interface/multi_window/restore_windows_on_load", true);
+	EDITOR_DEF("interface/multi_window/maximize_window", false);
```

Comment:
It's better to move these to `EditorSettings::_load_defaults`, so they can be picked by doctool.